packages/flutter/test/cupertino/colors_test.dart

```diff
@@ -0,0 +1,574 @@
+  normalColor: color0,
+  darkColor: color1,
+  elevatedColor: color2,
+  highContrastColor: color3,
+  darkElevatedColor: color4,
+  darkHighContrastColor: color5,
+  highContrastElevatedColor: color6,
+  darkHighContrastElevatedColor: color7,
+);
+
+final Color notSoDynamicColor1 = CupertinoDynamicColor(defaultColor: color0);
+final Color notSoDynamicColor2 = CupertinoDynamicColor(defaultColor: color0, darkColor: color0);
+final Color notSoDynamicColor3 = CupertinoDynamicColor(defaultColor: color0, highContrastColor: color0);
+final Color notSoDynamicColor4 = CupertinoDynamicColor(defaultColor: color0, elevatedColor: color0);
+final Color notSoDynamicColor5 = CupertinoDynamicColor(
+  normalColor: color0,
+  darkColor: color0,
+  darkHighContrastColor: color0,
+  darkElevatedColor: color0,
+  darkHighContrastElevatedColor: color0,
+  highContrastColor: color0,
+  highContrastElevatedColor: color0,
+  elevatedColor: color0,
+);
+
+final Color vibrancyDependentColor1 = CupertinoDynamicColor(
+  defaultColor: color1,
+  darkColor: color0,
+  darkHighContrastColor: color0,
+  darkElevatedColor: color0,
+  darkHighContrastElevatedColor: color0,
+);
+
+final Color contrastDependentColor1 = CupertinoDynamicColor(
+  defaultColor: color1,
+  highContrastColor: color0,
+  darkHighContrastColor: color0,
+  highContrastElevatedColor: color0,
+  darkHighContrastElevatedColor: color0,
+);
+
+final Color elevationDependentColor1 = CupertinoDynamicColor(
+  defaultColor: color1,
+  elevatedColor: color0,
+  darkElevatedColor: color0,
+  highContrastElevatedColor: color0,
+  darkHighContrastElevatedColor: color0,
+);
+
+void main() {
+  test('== works as expected', () {
+    expect(notSoDynamicColor1, notSoDynamicColor2);
+    expect(notSoDynamicColor1, notSoDynamicColor3);
+    expect(notSoDynamicColor1, notSoDynamicColor4);
+    expect(notSoDynamicColor1, notSoDynamicColor5);
+
+    expect(notSoDynamicColor2, notSoDynamicColor5);
+
+
+    expect(notSoDynamicColor1, isNot(vibrancyDependentColor1));
+    expect(notSoDynamicColor5, isNot(vibrancyDependentColor1));
+
+    expect(notSoDynamicColor1, isNot(contrastDependentColor1));
+    expect(notSoDynamicColor5, isNot(contrastDependentColor1));
+
+    expect(vibrancyDependentColor1, isNot(CupertinoDynamicColor(
+      defaultColor: color0,
+      darkColor: color0,
+      darkHighContrastColor: color0,
+      darkElevatedColor: color0,
+      darkHighContrastElevatedColor: color0,
+    )));
+  });
+
+  test('withVibrancy constructor works', () {
+    expect(vibrancyDependentColor1, CupertinoDynamicColor.withVibrancy(
+      defaultColor: color1,
+      normalColor: color1,
+      darkColor: color0,
+    ));
+  });
+
+  test('withVibrancyAndContrast constructor works', () {
+    expect(contrastDependentColor1, CupertinoDynamicColor.withVibrancyAndContrast(
+      defaultColor: color1,
+      highContrastColor: color0,
+      darkHighContrastColor: color0,
+    ));
+  });
```

Comment:
These above tests would be easier to parse with some comments or more detailed descriptions.  As it is now I have to scroll back and forth between the test and the constant colors declared above to figure out what exactly is being tested.

Reply:
Thanks for updating, it's a lot easier to understand now.